utils/scale-test

```diff
@@ -38,20 +38,23 @@ def has_debuginfo(swiftc):
 
 
 def write_input_file(args, ast, d, n):
-    ifile = os.path.join(d, "in%d.swift" % n)
-    with open(ifile,'w+') as f:
+    fname = "in%d.swift" % n
+    pathname = os.path.join(d, fname)
+    with open(pathname,'w+') as f:
         f.write(gyb.execute_template(ast, '', N=n))
-    return ifile
+    return fname
 
 
 def run_once_with_primary(args, ast, rng, primary_idx):
-    import sys, shutil, tempfile, json
+    import sys, tempfile, json
     r = {}
     try:
-        d = tempfile.mkdtemp()
+        if args.tmpdir != None and not os.path.exists(args.tmpdir):
```

Comment:
nit-pick: The linter will probably complain here as well. `None` is a singleton, the convention is to check for identity, not equality. So this should be `if args.tmpdir is not None`.